### PR TITLE
drivers: wifi: Set names to Wi-Fi related stacks

### PR DESCRIPTION
**Pull request by rado17**
Set names to Wi-Fi ralated stacks.

Fixes SHEL-2547

### CODE REVIEW

File: modules/hostap/src/supp_main.c [hunk 1]
```diff
@@ -553,6 +553,8 @@ static void z_wpas_start(void)
 					   CONFIG_WPA_SUPP_WQ_PRIORITY,
 					   NULL);
 
+	k_thread_name_set(&z_wpas_wq.thread, "supplicant");
```
Comment (krish2718) [inline modules/hostap/src/supp_main.c:556]: wpa_supplicant_wq

File: modules/hostap/src/supp_main.c [hunk 2]
```diff
@@ -623,6 +625,8 @@ static int z_wpas_init(void)
 			NULL, NULL, NULL,
 			0, 0, K_NO_WAIT);
 
+	k_thread_name_set(&z_wpa_s_tid, "supp_main");
```
Comment (krish2718) [inline modules/hostap/src/supp_main.c:628]: wpa_supplicant_main

File: drivers/wifi/nrf700x/src/work.c [hunk 3]
```diff
@@ -81,13 +81,17 @@ static int workqueue_init(void)
...
+
 	k_work_queue_init(&zep_wifi_intr_q);
 
 	k_work_queue_start(&zep_wifi_intr_q,
 						irq_wq_stack_area,
 						K_THREAD_STACK_SIZEOF(irq_wq_stack_area),
 						CONFIG_NRF700X_IRQ_WQ_PRIORITY,
 						NULL);
+
+	k_thread_name_set(&zep_wifi_intr_q.thread, "nrf700x_wifi_intrq");
```
Comment (krish2718) [inline drivers/wifi/nrf700x/src/work.c:94]:
```suggestion
	k_thread_name_set(&zep_wifi_intr_q.thread, "nrf700x_intr_wq");
```

File: drivers/wifi/nrf700x/src/work.c [hunk 4]
```diff
@@ -96,6 +100,8 @@ static int workqueue_init(void)
 						K_THREAD_STACK_SIZEOF(tx_done_wq_stack_area),
 						CONFIG_NRF700X_TX_DONE_WQ_PRIORITY,
 						NULL);
+
+	k_thread_name_set(&zep_wifi_tx_done_q.thread, "nrf700x_tx_done_q");
```
Comment (krish2718) [inline drivers/wifi/nrf700x/src/work.c:104]:
```suggestion
	k_thread_name_set(&zep_wifi_tx_done_q.thread, "nrf700x_tx_done_wq");
```

File: drivers/wifi/nrf700x/src/work.c [hunk 5]
```diff
@@ -106,6 +112,8 @@ static int workqueue_init(void)
 						K_THREAD_STACK_SIZEOF(rx_wq_stack_area),
 						CONFIG_NRF700X_RX_WQ_PRIORITY,
 						NULL);
+
+	k_thread_name_set(&zep_wifi_rx_q.thread, "nrf700x_rx_q");
```
Comment (krish2718) [inline drivers/wifi/nrf700x/src/work.c:116]:
```suggestion
	k_thread_name_set(&zep_wifi_rx_q.thread, "nrf700x_rx_wq");
```

File: modules/hostap/src/supp_main.c (hunk 1 above)
Comment (rado17) [reply modules/hostap/src/supp_main.c]: Done

File: modules/hostap/src/supp_main.c (hunk 2 above)
Comment (rado17) [reply modules/hostap/src/supp_main.c]: Done

File: drivers/wifi/nrf700x/src/work.c (hunk 3 above)
Comment (rado17) [reply drivers/wifi/nrf700x/src/work.c]: Done

File: drivers/wifi/nrf700x/src/work.c (hunk 4 above)
Comment (rado17) [reply drivers/wifi/nrf700x/src/work.c]: Done

File: drivers/wifi/nrf700x/src/work.c (hunk 5 above)
Comment (rado17) [reply drivers/wifi/nrf700x/src/work.c]: Done

**Review by jukkar — COMMENTED** [review]
Could you split the commit to two so that it is easier to cherry-pick just the module change in zephyr upstream.

**Review by sachinthegreen — APPROVED** [review]

**Review by kapbh — APPROVED** [review]

**Pull request merged**